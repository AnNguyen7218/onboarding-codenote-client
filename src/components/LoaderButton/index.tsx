import React from "react";
import { Button, Glyphicon } from "react-bootstrap";
import "./index.css";

interface IProps {
  isLoading: boolean,
  text: string,
  loadingText: string,
  className?: string,
  disabled?: boolean,
  [index: string]: any,
}

const LoaderButton = ({
  isLoading,
  text,
  loadingText,
  className = "",
  disabled = false,
  ...props
}: IProps) =>
  <Button
    className={`LoaderButton ${className}`}
    disabled={disabled || isLoading}
    {...props}
  >
    {isLoading && <Glyphicon glyph="refresh" className="spinning" />}
    {!isLoading ? text : loadingText}
  </Button>;

export default LoaderButton;
