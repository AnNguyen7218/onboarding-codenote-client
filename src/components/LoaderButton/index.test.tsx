import { shallow } from 'enzyme';
import React from 'react';
import LoaderButton from './index';

test('AppliedRoute renders without crashing', () => {
  let props
  const wrapper = shallow(<LoaderButton {...props} />);
  expect(wrapper).toMatchSnapshot();
});
