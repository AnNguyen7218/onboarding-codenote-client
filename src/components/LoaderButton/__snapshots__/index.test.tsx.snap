// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppliedRoute renders without crashing 1`] = `
<Button
  active={false}
  block={false}
  bsClass="btn"
  bsStyle="default"
  className="LoaderButton "
  disabled={false}
/>
`;
